src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -15,23 +15,136 @@
 
 #include "src/core/lib/event_engine/posix_engine/posix_engine.h"
 
+#include <atomic>
+#include <chrono>
 #include <string>
 #include <utility>
 
 #include "absl/container/flat_hash_set.h"
 #include "absl/functional/any_invocable.h"
+#include "absl/status/status.h"
 
 #include <grpc/event_engine/event_engine.h>
+#include <grpc/event_engine/memory_allocator.h>
 #include <grpc/support/log.h>
 
 #include "src/core/lib/debug/trace.h"
+#include "src/core/lib/event_engine/poller.h"
 #include "src/core/lib/event_engine/posix_engine/timer.h"
 #include "src/core/lib/event_engine/trace.h"
 #include "src/core/lib/event_engine/utils.h"
+#include "src/core/lib/experiments/experiments.h"
+#include "src/core/lib/gprpp/sync.h"
+
+#ifdef GRPC_POSIX_SOCKET_TCP
+#include "src/core/lib/event_engine/posix_engine/event_poller_posix_default.h"
+#endif  // GRPC_POSIX_SOCKET_TCP
+
+using namespace std::chrono_literals;
 
 namespace grpc_event_engine {
 namespace experimental {
 
+#ifdef GRPC_POSIX_SOCKET_TCP
+using grpc_event_engine::posix_engine::PosixEventPoller;
+
+PosixEnginePollerManager::PosixEnginePollerManager(
+    std::shared_ptr<ThreadPool> executor)
+    : poller_(grpc_event_engine::posix_engine::MakeDefaultPoller(this)),
+      executor_(std::move(executor)) {}
+
+PosixEnginePollerManager::PosixEnginePollerManager(PosixEventPoller* poller)
+    : poller_(poller),
+      poller_state_(PollerState::kExternal),
+      executor_(nullptr) {
+  GPR_DEBUG_ASSERT(poller_ != nullptr);
+}
+
+void PosixEnginePollerManager::Run(
+    experimental::EventEngine::Closure* closure) {
+  if (executor_ != nullptr) {
+    executor_->Run(closure);
+  }
+}
+
+void PosixEnginePollerManager::Run(absl::AnyInvocable<void()> cb) {
+  if (executor_ != nullptr) {
+    executor_->Run(std::move(cb));
+  }
+}
+
+void PosixEnginePollerManager::TriggerShutdown() {
+  // If the poller is external, dont try to shut it down. Otherwise
+  // set poller state to PollerState::kShuttingDown.
+  if (poller_state_.exchange(PollerState::kShuttingDown) ==
+      PollerState::kExternal) {
+    poller_ = nullptr;
+    return;
+  }
+
+  if (poller_ != nullptr) {
```

Comment:
I don't see any way that poller_ could be nullptr here?
Maybe we make it a `grpc_event_engine::posix_engine::PosixEventPoller* const poller_` in the declaration to enforce that it's initialized and avoid the need for the ifs.

Reply:
Removed the if statement. I couldn't make it a const though because the value needs to change a few lines above.